packages/kbn-eslint-plugin-eui-a11y/rules/eui_elements_should_have_a11y_props.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { Rule } from 'eslint';
+import { AST_NODE_TYPES, TSESTree, TSNode } from '@typescript-eslint/typescript-estree';
+
+import { checkNodeForExistingAriaLabelProp } from '../helpers/check_node_for_existing_data_test_subj_prop';
+import { getIntentFromNode } from '../helpers/get_intent_from_node';
+
+export const EUI_ELEMENTS = [
+  'EuiButtonIcon',
+  'EuiButtonEmpty',
+  'EuiBetaBadge',
+  'EuiSelect',
+  'EuiSelectWithWidth',
+];
+
+export const EuiElementsShouldHaveA11yProps: Rule.RuleModule = {
+  meta: {
+    type: 'suggestion',
+    fixable: 'code',
+  },
+  create(context) {
+    const { report, sourceCode } = context;
+
+    return {
+      JSXIdentifier: (node: TSESTree.Node) => {
+        if (!('name' in node)) {
+          return;
+        }
+
+        const name = String(node.name);
+        const range = node.range;
+        const parent = node.parent;
+
+        if (parent?.type !== AST_NODE_TYPES.JSXOpeningElement || !EUI_ELEMENTS.includes(name)) {
+          return;
+        }
+
+        const hasDataTestSubjProp = checkNodeForExistingAriaLabelProp(parent, () =>
+          // @ts-expect-error upgrade typescript v5.1.6
+          sourceCode.getScope(node as TSNode)
+        );
+
+        if (hasDataTestSubjProp) {
```

Comment:
Is this a valid case? Do we want to skip it if we have `data-test-subj` or am I missing something 🤔

Reply:
Variable name was left unadjusted when duplicating the existing ESLint custom rule package. Will double check the condition.